resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,307 @@
+if ([string]::IsNullOrEmpty($env:TestRegistryPath)) {
+    $global:tzAutoUpdatePath = 'HKLM:\SYSTEM\CurrentControlSet\Services\W32Time\Parameters'
+    $global:timeZoneInformationPath = 'HKLM:\SYSTEM\CurrentControlSet\Control\TimeZoneInformation'
+} else {
+    $global:tzAutoUpdatePath = $global:timeZoneInformationPath = $env:TestRegistryPath
+}
+
+#region Functions
+function TryGetRegistryValue {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )
+
+    if (Test-Path -Path $Key) {
+        try {
+            return (Get-ItemProperty -Path $Key -Name $Property -ErrorAction SilentlyContinue | Select-Object -ExpandProperty $Property)
+        } catch {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    } else {
+        Write-Verbose 'Registry key does not exist.'
+    }
+}
+function Get-ValidTimeZone {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $TimeZone
+    )
+
+    try {
+        $timeZoneId = (Get-TimeZone -Id $TimeZone -ErrorAction Stop).Id
+    } catch {
+        throw [System.Configuration.ConfigurationException]::new("Executing 'Get-TimeZone' failed. Error: $($PSItem.Exception.Message)")
+    }
+
+    return $timeZoneId
+}
```

Comment:
After rechecking your comment multiple times and looking at the code, I can see why I did it. Before this function, I was _enumming_ everything to allow users to always fill in a proper ID. I moved it to this, because it doesn't really make sense to test either `$true` or `$false` on this. If a user inserts an invalid ID, it should basically throw _somewhere_. Perhaps we can leave this function alone and let it throw when insert an invalid ID. In that way, when `Invoke-DscResource` is called, it returns a proper message. Let me know what  you think.